### PR TITLE
Remove py2 support from EG

### PR DESCRIPTION
**Pull request by rakshith91**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](../CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/eventgrid/azure-eventgrid/setup.py [hunk 1]
```diff
@@ -64,16 +64,14 @@
     classifiers=[
```
Comment (swathipil) [inline sdk/eventgrid/azure-eventgrid/setup.py:64]: lines 25-36 can be removed: https://github.com/Azure/azure-sdk-for-python/blob/main/doc/dev/packaging.md#note-on-checking-old-azure-packages

Comment (swathipil) [reply sdk/eventgrid/azure-eventgrid/setup.py:64]: oh interesting, I did not know. Maybe we can ask Laurent about this and send him a link to one of those questions?

File: sdk/eventgrid/azure-eventgrid/dev_requirements.txt [hunk 2]
```diff
@@ -4,5 +4,5 @@
 -e ../../identity/azure-identity
 -e ../azure-mgmt-eventgrid
 azure-storage-queue
-cloudevents<=2.0.0; python_version >= "3.6"
-aiohttp>=3.0; python_version >= '3.5'
+cloudevents<=2.0.0; python_version >= "3.7"
```
Comment (rakshith91) [inline sdk/eventgrid/azure-eventgrid/dev_requirements.txt:7]:
```suggestion
cloudevents<=2.0.0; python_version >= "3.6"
```

File: sdk/eventgrid/azure-eventgrid/setup.py [hunk 3]
```diff
@@ -64,16 +49,15 @@
...
-        'Programming Language :: Python :: 3',
+        'Programming Language :: Python :: 3 :: Only',
         'Programming Language :: Python :: 3.6',
         'Programming Language :: Python :: 3.7',
         'Programming Language :: Python :: 3.8',
         'Programming Language :: Python :: 3.9',
         'Programming Language :: Python :: 3.10',
         'License :: OSI Approved :: MIT License',
     ],
+    python_requires=">=3.7",
```
Comment (rakshith91) [inline sdk/eventgrid/azure-eventgrid/setup.py:60]:
```suggestion
    python_requires=">=3.6",
```

**Pull request merged**